src/components/layout/layout.tsx

```diff
@@ -1,12 +1,18 @@
-import Header from './header/header';
+import { Outlet, useLocation } from 'react-router-dom';
+import { ROUTE_PATH } from '@/constant/routes';
 import ProtectedRoute from '../common/protected-route/protected-route';
-
-import { Outlet } from 'react-router-dom';
+import Header from './header/header';
 
 const Layout = () => {
+  const { pathname } = useLocation();
+
+  // 현재 경로가 스터디룸 내부인지 확인하는 정규식
+  const regExp = new RegExp(`^${ROUTE_PATH.STUDY_ROOMS}/\\d+$`);
```

Comment:
정규식이 변수명으로만 표현하기 애매해서 주석을 붙여주신 것 같기는 한데요~
그래도 변수명으로 정규식을 표현해볼만한 것 같아요!

```suggestion
  const IN_STUDY_ROOM_REGEX = new RegExp(`^${ROUTE_PATH.STUDY_ROOMS}/\\d+$`);
```